### ISSUE TITLE
Thread Interrupted when Android app goes into the background

### ISSUE THREAD

**Issue by wardellbagby** [ReactiveX/RxJava#4459]
This issue affects both 1.x and 2.x.

When using RxJava and OkHttp in conjunction, while a network call is being made by OkHttp, if the app is put into the background, the thread used to observe, if it is one of the threads that the Schedulers class manages, will be interrupted. This issue doesn't occur if a thread is manually specified by using `Schedulers.from()`

I created a proof of concept Android app that showcases this issue here https://github.com/wardellchandler/RxJava-Bug

**Comment by akarnokd** [comment]
[Looks like you dispose](https://github.com/wardellchandler/RxJava-Bug/blob/master/app/src/main/java/com/mrchandler/rxbugtest/MainActivity.java#L85) the task on `onPause` - as most use cases would - which then promptly cancels the active sequences. I'm not experienced enough in Android (lifecycle) to give hints how to do background tasks that survive pause/resume.

**Comment by JakeWharton** [comment]
The common ways are one of two options:
- Move the request to not be bound directly to the UI (which should always be treated as transient), or
- Change the observable chain so that unsubscribing from the end doesn't cancel all the way upstream.

**Comment by wardellbagby** [comment]
@JakeWharton Can you elaborate more on how to do these two options?

**Comment by JakeWharton** [comment]
`.publish().autoConnect(1)` will prevent unsubscribing upstream when the downstream unsubscribes.

As to moving the call, you can use a singleton or service to host the network call and have your UI (aka the activity) just observe the data instead of trying to control it directly. There's a lot of different ways to do this so the scope of an example is a bit outside of this issue.

**Comment by wardellbagby** [comment]
@JakeWharton I was able to figure this out with your step in the right direction.

Using autoConnect(1) still had me running into the same problem; the thread would be interrupted everything had unsubscribed.

However, using autoConnect(0) had it working just fine, and it wasn't much different from the current system I was using; the Model would just be invoked slightly earlier. I pushed out changes to that effect. That could be seen here: https://github.com/wardellchandler/RxJava-Bug/commit/073dfa51c777c908def59d4bb314a643744966dc
